src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -231,6 +232,8 @@ private void collectUpdateSubmissionData(UpdateSubmission updateSubmission,
               .addAll(managingOfficerAdditionService.managingOfficerAdditions(submissionDto));
 
       ApiLogger.debug("Updates have been collected", logMap);
+    } else {
+      updateSubmission.setEntityName(submissionDto.getEntityName().getName());
```

Comment:
Out of curiosity, what is the reason for this to differ in the "change" and "no change" scenario? I would expect the same line and possibly even done in the `populateUpdateSubmission.populate()` method?

Reply:
It can't change in no change case but we have to get it from the public api in change case. Decided not to get it from the public API in no change case - as there is no need to do so.

Reply:
we have the fetched public data here but not in populate method (for change case).